engine/crates/runtime/src/kv.rs

```diff
@@ -60,5 +60,5 @@ impl std::ops::Deref for KvStore {
 #[async_trait::async_trait]
 pub trait KvStoreInner: Send + Sync {
     async fn get(&self, name: &str, cache_ttl: Option<Duration>) -> KvResult<Option<Vec<u8>>>;
-    async fn put(&self, name: &str, bytes: Vec<u8>, expiration_ttl: Option<Duration>) -> KvResult<()>;
+    async fn put(&self, name: &str, bytes: &[u8], expiration_ttl: Option<Duration>) -> KvResult<()>;
```

Comment:
`Cow<'_, [u8]>` would probably be the most flexible.

it's not great to use KV for caching. In Cloudflare KV behavior is different from cache. KV is more meant as a distributed KV than pure cache, so it'll try to fetch data from single colocation. Hence the `cache_ttl` in the `get()` to cache the result inside the current colocation. Entity cache shouldn't rely on that

Reply:
> Cow<'_, [u8]> would probably be the most flexible.

I guess, though I'm struggling to see a situation where it'd be useful?  An owned thing can always be made into a slice and a slice doesn't need converted?  Although let me know if I'm missing some advantage.

> it's not great to use KV for caching. In Cloudflare KV behavior is different from cache. KV is more meant as a distributed KV than pure cache, so it'll try to fetch data from single colocation. Hence the cache_ttl in the get() to cache the result inside the current colocation. Entity cache shouldn't rely on that

Raised #1938 to make this a bit more obvious.  I'll do something to fix this in a future PR: likely pulling in some kind of cache impl

Reply:
Ok flexible wasn't the right word for `Cow`, wanted to say it can avoid copies